src/js/tasks-function.js

```diff
@@ -0,0 +1,223 @@
+
+          // deleteBtn[elem.index].classList.add('active');
+          e.target.nextSibling.addEventListener('click', () => {
+            li[elem.index].remove();
+            this.removetask(elem.index);
+          });
+        });
+      });
+      li[elem.index].addEventListener('mouseleave', (e) => {
+        this.updateTask(e.target.id, e.target.innerText);
+      });
+      inputBox[elem.index].addEventListener('change', (e) => {
+        const status = new TaskStatus();
+        if (e.target.checked === true) {
+          status.checked(this.taskData[elem.index]);
+        } else {
+          status.unchecked(this.taskData[elem.index]);
+        }
+        this.updateTask(
+          e.target.nextSibling.id,
+          e.target.nextSibling.innerText,
+        );
+      });
+      if (this.taskData[elem.index].completed === true) {
+        inputBox[elem.index].setAttribute('checked', 'checked');
+        li[elem.index].classList.add('checked');
+        label[elem.index].style.textDecoration = 'line-through';
+      } else if (this.taskData[elem.index].completed === false) {
+        inputBox[elem.index].removeAttribute('checked');
+        label[elem.index].style.textDecoration = 'none';
+      }
+    });
+  };
+
+  addtask = (description, completed, index) => {
+    const newtask = new Tasks(description, completed, index);
+    this.taskData.push(newtask);
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('TodoListDB', JSON.stringify(this.taskData));
+    }
+  }
+
+  removetask = (item) => {
+    const key = item;
+    if (this.taskData.length === 1) {
+      this.taskData = [];
+    } else {
+      this.taskData.splice(key, 1);
+    }
+    this.taskData.forEach((elem, index) => {
+      elem.index = index;
+    });
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('TodoListDB', JSON.stringify(this.taskData));
+      this.displayTask();
+    }
+  }
+
+  updateTask(item, description) {
+    if (this.taskData[item]) {
+      this.taskData[item].description = description;
+    }
+    this.taskData.forEach((elem, index) => {
+      elem.index = index;
+    });
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('TodoListDB', JSON.stringify(this.taskData));
+      this.displayTask();
+    }
+  }
+
+  clearAllCompletTask = () => {
+    this.taskData = this.taskData.filter((elem) => elem.completed === false);
+    this.taskData.forEach((e, index) => {
+      e.index = index;
+    });
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('TodoListDB', JSON.stringify(this.taskData));
+      // eslint-disable-next-line no-unused-expressions
+      window.location.reload()();
+    }
+  }
+}
+
+// Add New Task and Display it
+let index = 0;
+const completed = false;
+const entryTask = new Todolist();
+
+const newItemInput = document.querySelector('#new-item');
+// const taskList = document.querySelector('.task-list');
```

Comment:
- [ ] Kindly delete any commented code as that makes your project look unprofessional.